server/src/main/java/org/opensearch/rest/RestController.java

```diff
@@ -395,6 +399,9 @@ private void tryAllHandlers(final RestRequest request, final RestChannel channel
                         return;
                     }
                 } else {
+                    if (!handleAuthenticateUser(request, channel)) {
```

Comment:
Should this be behind the feature flag?

Reply:
So I think it should be in public; however, the 'UnsupportedHttpAuthenticationToken' error should not be thrown, I've changed the error message to be logged at the debug level.

Reply:
I think this should be behind a feature flag as well. That way we don't to modify `extractToken` and `handleAuthenticateUser` to react appropriately when no auth token is found

Reply:
Scenarios like this are best expressed with object modeling, where functional differences are managed through the IdentityPlugin interfaces.  I've make sure the behavior with the NoopIdentityPlugin is identical to the previous behavior pattern - and the security plugin tests will quickly find any outliers.

Reply:
I've made this change inside the called method to keep this block of code easier to read